examples/vision/image_enhancer.py

```diff
@@ -0,0 +1,276 @@
+model performance helping us to assess the text information gain in real world scenarios
+when coupled with tesseract OCR engine.
+
+**References:**
+- [Enhancing OCR Accuracy with Super
+Resolution](https://cdn.iiit.ac.in/cdn/cvit.iiit.ac.in/images/ConferencePapers/2018/ocr_Ankit_Lat_ICPR_2018.pdf)
+Resolution](https://cdn.iiit.ac.in/cdn/cvit.iiit.ac.in/images/ConferencePapers/2018/ocr_Ankit_Lat_ICPR_2018.pdf)
+- [Improving the Perceptual Quality of Document
+Images Using Deep Neural
+Network](http://mile.ee.iisc.ac.in/publications/softCopy/DocumentAnalysis/ISNN_11page_65.pdf)
+Network](http://mile.ee.iisc.ac.in/publications/softCopy/DocumentAnalysis/ISNN_11page_65.pdf)
+"""
+
+"""
+## Additional Set-up
+"""
+
+# !pip install pybind11
+# !pip install fastwer
+# !pip install pytesseract
+# !sudo apt install tesseract-ocr
+
+import fastwer
+import pytesseract
+import numpy as np
+from glob import glob
+import tensorflow as tf
+from tensorflow import keras
+import matplotlib.pyplot as plt
+from tensorflow.keras import callbacks
+from tensorflow.keras.applications import resnet_v2
+
+"""
+## Dataset
+we have created a synthetic document dataset that depicts several real world document
+noises, we have used blurs and morphological filters to reconstruct real world noises.
+"""
+
+"""shell
+!gdown https://drive.google.com/uc?id=1jQfHJUUQcpktQcjIzitvb4SldKwqcAMr
+!unzip -q data.zip
+"""
+
+"""
+## Dataset Preprocessing
+"""
+
+train_source = sorted(glob("data/source/*.jpg"))[:-80]
+train_targets = sorted(glob("data/target/*.jpg"))[:-80]
+
+val_source = sorted(glob("data/source/*.jpg"))[-80:]
+val_targets = sorted(glob("data/target/*.jpg"))[-80:]
+
+height, width = 352, 608
+batch_size = 4
+
+
+def resize_with_pad(image, height, width):
+    return tf.image.resize_with_pad(
+        image=image, target_height=height, target_width=width
+    )
+
+
+def data_preprocess(source, target):
+    source_image = tf.io.decode_png(tf.io.read_file(source), channels=3)
+    target_image = tf.io.decode_png(tf.io.read_file(target), channels=3)
+    source_image = resize_with_pad(source_image, height=height, width=width)
+    target_image = resize_with_pad(target_image, height=height, width=width)
+    source_image = resnet_v2.preprocess_input(source_image)
+    target_image = resnet_v2.preprocess_input(target_image)
+    return source_image, target_image
+
+
+buffer_size = int(len(train_source) * 0.2)
+train_pipe = tf.data.Dataset.from_tensor_slices((train_source, train_targets))
+train_pipe = train_pipe.map(data_preprocess, tf.data.AUTOTUNE).cache()
+train_pipe = train_pipe.shuffle(buffer_size).batch(batch_size).repeat()
+
+
+val_pipe = tf.data.Dataset.from_tensor_slices((val_source, val_targets))
+val_pipe = val_pipe.map(data_preprocess, tf.data.AUTOTUNE).cache()
+val_pipe = val_pipe.batch(batch_size).repeat()
+
+
+train_step = len(train_source) // batch_size
+val_step = len(val_source) // batch_size
```

Comment:
These two variables are only needed in `model.fit`. I think it's better to state in the training section. Also, you can simply just pass them to fit function as they are not used anywhere else.